http-clients/src/main/java/feign/jaxrs/JAXRSWithHeaderAndQueryMapContract.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feign.jaxrs;
+
+import feign.Contract;
+import feign.HeaderMap;
+import feign.MethodMetadata;
+import feign.QueryMap;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+/**
+ * Contract that provides the functionality of {@link JAXRSContract} with full support for
+ * the {@link QueryMap} annotation. Allows the QueryMap annotation to be used in methods
+ * that also take a body parameter (such as {@link javax.ws.rs.POST or {@link javax.ws.rs.PUT}}).
+ */
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public final class JAXRSWithHeaderAndQueryMapContract extends Contract.BaseContract {
```

Comment:
OK will go with that -- was doing `JAXRS` to match what Feign does, but since this is one of our classes fine with using our conventions